samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -0,0 +1,598 @@
+                        Write-Host "Added Users: $Emails."
+                    }
+                }
+
+                if ($Schedules) {
+                    $Schedules | ForEach-Object { $_ | New-AzLabSchedule -Lab $lab } | Out-Null
+                    Write-Host "Added all schedules."
+                }
+
+                Write-Host "Completed creation of $LabName, total duration $(((Get-Date) - $StartTime).TotalSeconds) seconds" -ForegroundColor Green
+            }
+        }
+
+        # No need to parallelize this one as super fast
+        function New-ResourceGroups {
+            [CmdletBinding()]
+            param(
+                [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+                [psobject[]]
+                $ConfigObject
+            )
+
+            $Rgs = $ConfigObject | Select-Object -Property ResourceGroupName, Location -Unique
+            Write-Host "Operating on the following RGs:"
+            Write-Host $Rgs
+            
+            $Rgs | ForEach-Object {
+                if (-not (Get-AzResourceGroup -ResourceGroupName $_.ResourceGroupName -EA SilentlyContinue)) {
+                    New-AzResourceGroup -ResourceGroupName $_.ResourceGroupName -Location $_.Location | Out-null
+                    Write-Host "$($_.ResourceGroupName) resource group didn't exist. Created it."
+                }
+            }
+        }
+
+        function New-Accounts {
+            [CmdletBinding()]
+            param(
+                [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+                [psobject[]]
+                $ConfigObject
+            )
+
+            $lacs = $ConfigObject | Select-Object -Property ResourceGroupName, LabAccountName -Unique
+            Write-Host "Operating on the following Lab Accounts:"
+            Write-Host $lacs
+
+            $block = {
+                param($path, $ResourceGroupName, $LabAccountName)
+
+                Set-StrictMode -Version Latest
+                $ErrorActionPreference = 'Stop'
+
+                $modulePath = Join-Path $path '..\Az.LabServices.psm1'
+                Import-Module $modulePath
+
+                if ((Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName) -eq $null ){
+                    New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                }
+                Write-Host "$LabAccountName lab account created or found."
+            }
+            
+            Write-Host "Starting lab accounts creation in parallel. Can take a while."
+            $jobs = @()
+            $lacs | ForEach-Object {
+                $jobs += Start-ThreadJob -ScriptBlock $block -ArgumentList $PSScriptRoot, $_.ResourceGroupName, $_.LabAccountName -Name $_.LabAccountName -ThrottleLimit $ThrottleLimit
+            }
+
+            $hours = 1
+            $jobs | Wait-Job -Timeout (60 * 60 * $hours) | Receive-Job
```

Comment:
Using this syntax (in the pipeline) the output for multiple jobs can get jumbled together.  Better to use a ForEach-Object to process the output in order, Line 404 has a good pattern to use.